### PR TITLE
[TS] LPS-205139

### PR DESCRIPTION
**Pull request by marianoalvarosaiz**

### CODE REVIEW

**Comment by marianoalvarosaiz** [comment]
ci:test:sf

**Comment by marianoalvarosaiz** [comment]
@yuhai done and force-pushed.

File: modules/util/portal-tools-service-builder/bnd.bnd [hunk 1]
```diff
@@ -53,6 +53,7 @@ Main-Class: com.liferay.portal.tools.service.builder.ServiceBuilder
 	@com.liferay.petra.sql.dsl.api-*.jar!/com/liferay/petra/sql/dsl/query/sort/OrderByInfo.class,\
 	@com.liferay.petra.string-*.jar!/com/liferay/petra/string/*.class,\
 	@com.liferay.portal.impl-*.jar!/com/liferay/portal/model/BaseModelHintsImpl.class,\
+	@com.liferay.portal.impl-*.jar!/com/liferay/portal/model/BaseModelHintsImpl$FieldDataBag.class,\
 	@com.liferay.portal.impl-*.jar!/com/liferay/portal/service.xml,\
 	@com.liferay.portal.impl-*.jar!/com/liferay/portal/tools/ArgumentsMap.class,\
```
Comment (yuhai) [inline modules/util/portal-tools-service-builder/bnd.bnd:58]: @marianoalvarosaiz, I just found `BaseModelHintsImpl$1.class` didn't been include to the module. Please change it from
`@com.liferay.portal.impl-*.jar!/com/liferay/portal/model/BaseModelHintsImpl$FieldDataBag.class,\ ` to 
`@com.liferay.portal.impl-*.jar!/com/liferay/portal/model/BaseModelHintsImpl$*.class,\`

**Comment by tinatian** [comment]
@marianoalvarosaiz I have a concern when you change portal impl version from 2.71.9 to default. Service builder is tool to be used by all versions of portal, we expected it to be as stable as possible. By using default, each time you build a service builder, you may end up including java files with different content which may cause different behaviors.

I am wondering what exactly service builder is affected by this bug? can we fix the bug by avoiding usages of ToolsUtil? or maybe inline the used logic of ToolsUtil is also an option.

Tina.

**Comment by tinatian** [comment]
@marianoalvarosaiz I am not that worried about using default for com.liferay.portal.kernel as we know com.liferay.portal.kernel most time include interfaces, not too many implementations, while we do a lot of real impl logic in portal-impl.

I bring this concern up as it did happen before, I have blur memory that Brian does not want us to do that.

If we have to change the version we used, I would rather to change JavaParser rather than portal-impl, as JavaParser is separated and provides single feature, while we have so many different stuff in portal-impl.

Could you please try some other solutions? Also inline the used api into service builder is also an option if not too much.

Thanks.
Tina.

**Comment by marianoalvarosaiz** [comment]
@tinatian see https://github.com/liferay-core-infra/liferay-portal/pull/6489

**Pull request closed without merging**